### PR TITLE
answers for lesson 8

### PR DESCRIPTION
**Pull request by ElenaCherpakova**

### CODE REVIEW

File: 05-JWT-Basics/preferred/controllers/main.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    expiresIn: '24h',
+  });
+  res.status(200).json({ msg: 'user created', token });
+};
+
+const hello = async (req, res) => {
+  let { name } = req.user;
+  res.status(200).json({
+    msg: `Hello, ${name}!`,
+  });
```
Comment (jdevries3133) [inline 05-JWT-Basics/preferred/controllers/main.js:18]: If the name comes from the users request is it possible for the user to lie about their name?

File: 05-JWT-Basics/preferred/middleware/auth.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+const authenticationMiddleware = async (req, res, next) => {
+  const authHeader = req.headers.authorization;
+  if (!authHeader || !authHeader.startsWith('Bearer ')) {
+    return res.status(401).json({
+      message: 'No token provided',
+    });
+  }
+  const token = authHeader.split(' ')[1];
+  try {
+    const decoded = jwt.verify(token, process.env.JWT_SECRET);
```
Comment (jdevries3133) [inline 05-JWT-Basics/preferred/middleware/auth.js:12]: What are we verifying?

**Review by jdevries3133 — APPROVED** [review]
Nice work!

File: 05-JWT-Basics/preferred/middleware/auth.js (hunk 2 above)
Comment (ElenaCherpakova) [reply 05-JWT-Basics/preferred/middleware/auth.js:12]: This middleware function checks if a user is authorized to access protected resources by verifying the authenticity of a JWT token.

File: 05-JWT-Basics/preferred/controllers/main.js (hunk 1 above)
Comment (ElenaCherpakova) [reply 05-JWT-Basics/preferred/controllers/main.js:18]: Yes, if the name comes from the user's request and we don't have any validation set up it is possible for the user to lie about their name.

**Comment by ElenaCherpakova** [comment]
Thank you for your questions, Jack.

File: 05-JWT-Basics/preferred/controllers/main.js (hunk 1 above)
Comment (jdevries3133) [reply 05-JWT-Basics/preferred/controllers/main.js:18]: Does the name come from the users request in this case?

File: 05-JWT-Basics/preferred/middleware/auth.js (hunk 2 above)
Comment (jdevries3133) [reply 05-JWT-Basics/preferred/middleware/auth.js:12]: > verifying the authenticity of a JWT token.

How do we verify the authenticity of a JWT?

Comment (ElenaCherpakova) [reply 05-JWT-Basics/preferred/middleware/auth.js:12]: We use the ```jwt.verify()``` method from the ```jsonwebtoken``` library. This method takes two arguments: the token to be verified and a secret key to decode the signature. The token we extract from ```req.header.authorization```.

File: 05-JWT-Basics/preferred/controllers/main.js (hunk 1 above)
Comment (ElenaCherpakova) [reply 05-JWT-Basics/preferred/controllers/main.js:18]: The name value is coming from a verified and authenticated JWT included in the user's request. There is a reason why we store the name value in the ```req.user``` object - this is a standard way to pass data between middleware functions and route handlers.

**Comment by jdevries3133** [comment]
Great understand of JWTs 👍

**Pull request merged**